src/mongocxx/exception/error_code.cpp

```diff
@@ -54,7 +54,7 @@ class error_category final : public std::error_category {
             case error_code::k_server_response_malformed:
                 return "the response from the server was malformed";
             case error_code::k_invalid_uri:
-                return "an invalid MongoDB URI was provided";
+                return "invalid URI schema, expecting 'mongodb://' or 'mongodb+srv://'";
```

Comment:
Why this change?

Reply:
Based on the description of the ticket, I assumed that we wanted the warning generated by the C driver to be used as the error message when throwing the exception (as well as provides additional context to the "invalid URI" error). Should we leave it to how it was before?

Reply:
Your assumption is right! The error message here before was for the general `k_invalid_uri`  error code. The text here now refers to a specific error code. Let's keep this general and add the error message (`error.message`) returned from the _C_ function as an argument to the `logic_error` constructor. So instead of:
```
        throw logic_error{error_code::k_invalid_uri};
```
We can just do this:
```
        throw logic_error{error_code::k_invalid_uri, error.message};
```